skimage/exposure/histogram_matching.py

```diff
@@ -8,17 +8,27 @@ def _match_cumulative_cdf(source, template):
     Return modified source array so that the cumulative density function of
     its values matches the cumulative density function of the template.
     """
-    src_values, src_unique_indices, src_counts = np.unique(source.ravel(),
-                                                           return_inverse=True,
-                                                           return_counts=True)
-    tmpl_values, tmpl_counts = np.unique(template.ravel(), return_counts=True)
+    if source.dtype.kind == 'u':
+        src_lookup = source.ravel()
+        src_counts = np.bincount(src_lookup)
+        tmpl_counts = np.bincount(template.ravel())
```

Comment:
```suggestion
        src_lookup = source.reshape(-1)
        src_counts = np.bincount(src_lookup)
        tmpl_counts = np.bincount(template.reshape(-1))
```

I think @rfezzani typically encourages me to use `reshape(-1)` over ravel due to the following statement in the NumPy ravel docstring:
`When a view is desired in as many cases as possible, arr.reshape(-1) may be preferable.`